src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSetSerializer.cs

```diff
@@ -151,9 +154,9 @@ public override void WriteObject(object graph, Type type, ODataMessageWriter mes
             // object before calling WriteEnd(), the next page link will be written at the end, as required for
             // odata.streaming=true support.
 
-            if (nextPageLink != null)
+            if (nextLinkGenerator != null)
```

Comment:
>nextLinkGenerator != null [](start = 16, length = 25)

Again, I think you want to check nextLinkGenerator(lastMember) != null

Reply:
Fixed this. This was accurate as of a previous implementation but you are right, nextLinkGenerator cannot be null atleast in the default implementation. 


---
In reply to: [248349442](https://github.com/OData/WebApi/pull/1724#discussion_r248349442) [](ancestors = 248349442)